src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs

```diff
@@ -42,10 +42,10 @@ public partial class NodeAutoCompleteSearchControl : IDisposable
         public NodeAutoCompleteSearchControl()
         {
             InitializeComponent();
-            if (Application.Current != null)
+            if (string.IsNullOrEmpty(ViewModel?.dynamoViewModel.Model.HostAnalyticsInfo.HostName) && Application.Current != null)
```

Comment:
hi @zeusongit Thinking about this again, do we need to touch this logic? if we do so, these code will not be called in Sandbox case right?

Reply:
I think we either all need to agree on what the strings will be and what they mean (using enums) or we'll need to come up with a better heuristic. 

Also please note @BogdanZavu's PR where some of these become static, and @sm6srw 's PR where some of these disappear, it will need to be resolved if we proceed with this approach.

Reply:
I think we should wait for their PRs to be merged then we can resolve the remaining instances of Application.Current if need be.

Reply:
please discuss timing with @sm6srw - I think he is also waiting. A deadlock!

Reply:
@sm6srw any idea when your PR to remove obsolete code will be ready to merge?